app/actions/websocket/posts.ts

```diff
@@ -0,0 +1,191 @@
+    //     const dmAction = makeDirectChannelVisibleIfNecessary(state, otherUserId);
+    //     if (dmAction) {
+    //         actions.push(dmAction);
+    //     }
+    // } else if (msg.data.channel_type === General.GM_CHANNEL) {
+    //     const gmActions = await makeGroupMessageVisibleIfNecessary(state, post.channel_id);
+    //     if (gmActions) {
+    //         actions.push(...gmActions);
+    //     }
+    // }
+
+    //const viewingGlobalThreads = getViewingGlobalThreads(state);
+    // const collapsedThreadsEnabled = isCollapsedThreadsEnabled(state);
+    // actions.push(receivedNewPost(post, collapsedThreadsEnabled));
+    if (!shouldIgnorePost(post)) {
+        let markAsRead = false;
+        let markAsReadOnServer = false;
+
+        if (!myChannel?.manuallyUnread) {
+            if (
+                post.user_id === currentUserId &&
+                !isSystemMessage(post) &&
+                !isFromWebhook(post)
+            ) {
+                markAsRead = true;
+                markAsReadOnServer = false;
+            } else if ((post.channel_id === currentChannelId)) { // TODO: THREADS && !viewingGlobalThreads) {
+                // Don't mark as read if we're in global threads screen
+                // the currentChannelId still refers to previously viewed channel
+                markAsRead = true;
+                markAsReadOnServer = true;
+            }
+        }
+
+        if (markAsRead && markAsReadOnServer) {
+            markChannelAsRead(serverUrl, post.channel_id);
+        } else if (markAsRead) {
+            const viewedAt = await markChannelAsViewed(serverUrl, post.channel_id, true);
+            if (viewedAt instanceof Model) {
+                models.push(viewedAt);
+            }
+        } else {
+            const unreadAt = markChannelAsUnread(serverUrl, post.channel_id, 1, msg.data.mentions, false, true);
+            if (unreadAt instanceof Model) {
+                models.push(unreadAt);
+            }
+        }
+    }
+
+    database.operator.batchRecords(models);
+}
+
+export async function handlePostEdited(serverUrl: string, msg: WebSocketMessage) {
+    const database = DatabaseManager.serverDatabases[serverUrl];
+    if (!database) {
+        return;
+    }
+    const currentUserId = await queryCurrentUserId(database.database);
+    const data = JSON.parse(msg.data.post);
+    const post = {
+        ...data,
+        ownPost: data.user_id === currentUserId,
+    };
+
+    try {
+        fetchPostAuthors(serverUrl, [post]);
+    } catch {
+        // Do nothing
+    }
+
+    database.operator.handlePosts({
+        actionType: ActionType.POSTS.RECEIVED_NEW,
+        order: [post.id],
+        posts: [post],
+    });
+}
+
+export function handlePostDeleted(serverUrl: string, msg: WebSocketMessage) {
+    const data = JSON.parse(msg.data.post);
+
+    removePostById(serverUrl, data.id);
+}
+
+export async function handlePostUnread(serverUrl: string, msg: WebSocketMessage) {
```

Comment:
I need the channels to be working to properly test this.

Reply:
it should be fine checking the database data ;)